packages/story-editor/src/components/panels/design/textAccessibility/textAccessibility.js

```diff
@@ -32,61 +32,47 @@ import { useState, useEffect } from '@googleforcreators/react';
  * Internal dependencies
  */
 import { Row } from '../../../form';
+import { getCommonValue } from '../../shared';
 import { SimplePanel } from '../../panel';
 import { useHighlights, states, styles } from '../../../../app/highlights';
 import { MULTIPLE_VALUE, MULTIPLE_DISPLAY_VALUE } from '../../../../constants';
 import { combineElementsWithTags } from './utils';
 
-// Options to be mapped for the dropdown
-const semanticHeadingsOptionsMap = [
-  {
-    label: __('Heading 1', 'web-stories'),
-    value: 'h1',
-  },
-  {
-    label: __('Heading 2', 'web-stories'),
-    value: 'h2',
-  },
-  {
-    label: __('Heading 3', 'web-stories'),
-    value: 'h3',
-  },
-  {
-    label: __('Paragraph', 'web-stories'),
-    value: 'p',
-  },
-  {
-    label: __('Automatic', 'web-stories'),
-    value: 'auto',
-  },
-];
+const HEADING_LEVELS = {
+  h1: __('Heading 1', 'web-stories'),
+  h2: __('Heading 2', 'web-stories'),
+  h3: __('Heading 3', 'web-stories'),
+  p: __('Paragraph', 'web-stories'),
+};
 
 function TextAccessibilityPanel({ selectedElements, pushUpdate }) {
   // Feature flagging for Semantic Headings
   const showSemanticHeadings = useFeature('showSemanticHeadings');
   const [selectedTextElements, setSelectedTextElements] = useState([]);
   const [currentTags, setCurrentTags] = useState([]);
```

Comment:
I don't 100% understand why these local states are necessary at all looking closer into this component? What state is required, that isn't already stored in the story reducer?

Reply:
@barklund Please note that this PR just adds a few suggestions to the original PR. I recommend reviewing #11888 instead

Reply:
> 

Indeed, this is only an addition to that PR, but I can briefly explain:
Local state is needed because tags and selected elements, by default, are not combined in the story reducer. We also need to separately know how many tags they are, and whether they match, to support multiple text items being selected and proper display text (if a H1 and an H2 are selected, we say 'mixed' for the heading levels. if two H2s are selected, we show that both elements are 'Heading 2'). For that reason, both selected elements with their tags, as well as the original list of tags, are added to component state.